src/python/pants/engine/internals/graph.py

```diff
@@ -382,7 +382,9 @@ async def coarsened_targets(addresses: Addresses) -> CoarsenedTargets:
         if not any(component_address in addresses_set for component_address in component):
             continue
         component_set = set(component)
-        members = tuple(addresses_to_targets[a] for a in component)
+        members = tuple(
```

Comment:
what was broken here? 👀

Reply:
Even if the `scc` algorithm is stable, I don't really trust that the inputs will be stable (for example, if we enter the transitive graph at a different spot, etc). So mostly a precaution to uphold the deterministic member ordering guarantee for `CoarsenedTarget`.